src/index.ts

```diff
@@ -1,130 +1,234 @@
-                fallBackToAapt = true
-              },
-              () => subscriber.complete(),
-            )
-          }),
-        },
-        {
-          title: chalk`Encoding using AAPT {dim [fallback]}`,
-          skip: () => !fallBackToAapt,
-          task: () => apktool.encode(decodeDir, unsignedApkPath, false),
-        }
-      ]),
+      title: "Encoding patched APK file",
+      task: () =>
+        new Listr([
+          {
+            title: "Encoding using AAPT2",
+            task: (_, task) =>
+              new Observable(subscriber => {
+                apktool.encode(decodeDir, unsignedApkPath, true).subscribe(
+                  line => subscriber.next(line),
+                  () => {
+                    subscriber.complete();
+                    task.skip("Failed, falling back to AAPT...");
+                    fallBackToAapt = true;
+                  },
+                  () => subscriber.complete()
+                );
+              })
+          },
+          {
+            title: chalk`Encoding using AAPT {dim [fallback]}`,
+            skip: () => !fallBackToAapt,
+            task: () => apktool.encode(decodeDir, unsignedApkPath, false)
+          }
+        ])
     },
     {
-      title: 'Signing patched APK file',
-      task: () => new Observable(subscriber => {
-        (async () => {
-          await uberApkSigner.sign(unsignedApkPath)
-            .forEach(line => subscriber.next(line))
-
-          await fs.copyFile(
-            path.join(tmpDir, 'unsigned-aligned-debugSigned.apk'),
-            finishedApkPath,
-          )
-
-          subscriber.complete()
-        })()
-      }),
-    },
-  ]).run().catch(error => {
-    console.error(
-      chalk`\n  {red.inverse.bold  Failed! } An error occurred:\n\n`,
-      error.toString(),
-    )
-
-    process.exit(1)
-  })
+      title: "Signing patched APK file",
+      task: () =>
+        new Observable(subscriber => {
+          (async () => {
+            await uberApkSigner
+              .sign(unsignedApkPath)
+              .forEach(line => subscriber.next(line));
+
+            await fs.copyFile(
+              path.join(tmpDir, "unsigned-aligned-debugSigned.apk"),
+              finishedApkPath
+            );
+
+            subscriber.complete();
+          })();
+        })
+    }
+  ])
+    .run()
+    .catch(error => {
+      console.error(
+        chalk`\n  {red.inverse.bold  Failed! } An error occurred:\n\n`,
+        error.toString()
+      );
+
+      process.exit(1);
+    });
 
   if (usesAppBundle) {
     console.log(
       chalk`{yellow
   {inverse.bold  WARNING }
 
   This app seems to be using {bold Android App Bundle} which is dependent on Google Play.
-  As a result, installing it through an APK is probably {bold not going to work},
-  regardless of whether it has been patched by apk-mitm or not.}`
+  This feature is very {bold experimental} as a result {bold it might not work}`
```

Comment:
As I already said in my other comment, we should tell the user to acquire or export a bundle file (`.xapk` or `.apks`) here and run `apk-mitm` again instead of proceeding. You also need to add the closing bracket for the `{yellow` section back in.

Reply:
Done ✅